### PR TITLE
Fixing test, adding parameters to blastn, minor bugfixes

### PR DESCRIPTION
**Pull request by jannikseidelQBiC**
# Blastn parameters
To reduce the size of output files from blastn `evalue, max_target_seqs and max_hsps` were introduced as parameters. 

# test.conf
To avoid failing the minimal test because of the `tax2filter` (taxon to filter, at the moment only checking if the taxon is in the data) not being in the used minigut kraken2 db it was changed to "unclassified".

# test_full.conf
To make a reasonable test only the 8 GB Standard kraken2 is now used instead of Standard (approx. 60 GB).

# fasta parameter for the creation of blastn db
To avoid failure of the pipeline because of an empty `fasta` parameter (which should be taken from the `igenome.conf` in this case) it was moved to the `workflow/detaxizer.nf`.

# Samplesheets
The `samplesheet.csv` for the minimal test was reduced so that a small test can be run by github
The benchmarking samplesheet was expanded with testdata for a mix of human and Covid19 data.
The long reads were excluded for now and will be added later on.

# `modules/local/parse_kraken2report.nf` 
The pipeline now fails gracefully if the `tax2filter` is not in the kraken2 DB and shows an explanatory error message.

# `prepare_fasta4blastn.nf` now supports docker
Docker support was added to make tests possible.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/detaxizer/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/detaxizer _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/test_full.config [hunk 1]
```diff
@@ -21,10 +21,10 @@ params {
...
     fasta         = "s3://ngi-igenomes/igenomes/Homo_sapiens/NCBI/GRCh38/Sequence/WholeGenomeFasta/genome.fa"
     // Kraken2 test db
-    kraken2db        = "https://genome-idx.s3.amazonaws.com/kraken/k2_standard_20230605.tar.gz"
+    kraken2db           = "https://genome-idx.s3.amazonaws.com/kraken/k2_standard_08gb_20231009.tar.gz"
     kraken2confidence   = 0.00
     tax2filter          = 'Mammalia'
 
     blast_coverage      = 40.0
-    blast_similarity    = 40.0
+    blast_identity    = 40.0
```
Comment (d4straub) [inline conf/test_full.config:29]: alignment imperfect ;)

File: modules/local/parse_kraken2report.nf [hunk 2]
```diff
@@ -122,7 +122,12 @@ kraken_dict = kraken_taxonomy2hierarchy(kraken_list)
 list_kraken = get_all_keys(kraken_dict)
 result = remove_incomplete_taxa(list_kraken)
 result = generate_dict_for_lookup(result)
-result_to_filter = result["$params.tax2filter"] + ["$params.tax2filter"]
+try:
```
Comment (d4straub) [inline modules/local/parse_kraken2report.nf:125]: In the whole file seems to be something off with the indentation I think. One indent missing?

File: nextflow.config [hunk 3]
```diff
@@ -35,7 +35,7 @@ params {
     fastp_cut_mean_quality               = 15
     save_clipped_reads                   = false
 
-    // Taxon to check for/to filter in kraken2 format <taxid XYZ>
+    // Taxon to check for/to filter in kraken2 format, it has to be in the used kraken2 database
```
Comment (d4straub) [inline nextflow.config:38]:
```suggestion
    // Taxon to check for/to filter in kraken2 format, it has to be present in the supplied kraken2 database
```

**Review by d4straub — COMMENTED** [review]
I think its fine as soon as EC linting & prettier pass.
Indentation seems off sometimes, would be good to unify.

**Comment by jannikseidelQBiC** [comment]
@nf-core-bot fix linting

**Review by d4straub — APPROVED** [review]
Looks good!

**Review by d4straub — APPROVED** [review]
LGTM

**Comment by ewels** [comment]
The repo has not been set up properly. I've changed the access settings, your review is all green now ✅ 

I'll run the scripts to configure the rest of the settings now.

**Comment by d4straub** [comment]
Thanks!

**Comment by d4straub** [comment]
Could someone with admin right fix the expected (but never executed) test "Run pipeline with test data"? @FriederikeHanssen ?

**Comment by mashehu** [comment]
done

**Pull request merged**